### ISSUE TITLE
Is the `@dialyzer` module attribute suggestion for `Phoenix.Presence` in the Wiki still accurate?

### ISSUE THREAD

**Issue by bmitc** [jeremyjh/dialyxir#459]
### Precheck

* I performed a search but didn't see anything.
* We unfortunately cannot upgrade Dialyxir right now as we are in the middle of staging several version upgrades, starting with Elixir and Erlang. But we will be doing so soon.

### Environment

* Elixir & Erlang/OTP versions (elixir --version): Elixir 1.13.4 and Erlang/OTP 24.3.3

* Which version of Dialyxir are you using? (cat mix.lock | grep dialyxir): 0.5.1

Note: As mentioned above, we are in the process of upgrading versions of Elixir and Erlang/OTP and will then follow-up shortly with dependencies. So we will be upgrading Dialyxir, just not right now.

### Current behavior

In the [`Phoenix Dialyxir Quickstart`](https://github.com/jeremyjh/dialyxir/wiki/Phoenix-Dialyxir-Quickstart) Wiki entry, it is mentioned that for `Phoenix.Presence`, we should have:
```elixir
  @dialyzer [
    {:nowarn_function, 'init': 1},
    {:nowarn_function, 'track': 3},
    {:nowarn_function, 'track': 4},
    {:nowarn_function, 'update': 3},
    {:nowarn_function, 'update': 4},
    ]
```

After upgrading from Elixir 1.11.4 and Erlang/OTP 23.0.2 to Elixir 1.13.4 and Erlang/OTP 24.3.3, I know receiving a compiler error stating:

```
** (CompileError) lib/apm_web/channels/presence.ex:1: undefined function init/1 given to @dialyzer :nowarn_function
```

This compile error was not generated before the upgrade, and I haven't touched any other dependency.

### Expected behavior
I am new to Phoenix, and as far as I can tell, `init/1` is and was not a callback in the `Phoenix.Presence` module.

I have two questions:

1. Does the Wiki article need to be updated to remove `init/1`?
    a. If not, where should this function be coming from?
2. Any thoughts about why the Elixir/Erlang upgrade caused this to go from no compile error to a compile error?

Thank you!

**Comment by bmitc** [comment]
Tagging @OvermindDL1 since they added the `Phoenix.Presence` section [here](https://github.com/jeremyjh/dialyxir/wiki/Phoenix-Dialyxir-Quickstart/56c936f3a34da6c72024bf84591361ca492009a1) and @jeremyjh since they seem to be the primary author of that Wiki page (I suppose redundant give their status as owner of this repository).

**Comment by bmitc** [comment]
Thanks for taking a look @jeremyjh! What was curious to me was that I didn't update Phoenix. I only updated Elixir and Erlang. So maybe some macro shenanigans in Phoenix no longer expose whatever the `init/1` function is/was due to a change in Elixir?

I am not overly concerned but was just trying to understand why the version bump on Elixir and Erlang caused this to become a compile error. Getting rid of `{:nowarn_function, 'init': 1}` got rid of the compiler error.
